core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -171,5 +172,9 @@ object Consumer {
     }
   }
 
+  def committablePartitionedSource[K, V](settings: ConsumerSettings[K, V]):
+    Source[(TopicPartition, Source[CommittableMessage[K, V], Control]), Control] = {
```

Comment:
I tried to summarize what we want to achieve here - https://github.com/akka/reactive-kafka/issues/93#issuecomment-209861190

I read the explanation and still thinking that `Source[(Topic, Source)]` is viable option. All resources managed from main `Source` and substreams can be explicitly closed via `control.shutdown`. Yes we still can have potentially deadlock, but numbers of partitions are relatively small. 

Other approach is to use SubStreams, but I can not figure out how can we allow to perform a user code in reaction to a partition assignment.

Any other ideas welcome!